recipe/meta.yaml

```diff
@@ -36,42 +36,31 @@ outputs:
         - pip
         - python
       run:
-        - alembic  # [mlflow_variant != "skinny"]
-        - click >=7.0
-        - cloudpickle
-        - databricks-cli >=0.8.7
-        - docker-py >=4.0.0  # [mlflow_variant != "skinny"]
-        - entrypoints
-        - flask  # [mlflow_variant != "skinny"]
-        - gitpython >=2.1.0
+        - alembic <2  # [mlflow_variant != "skinny"]
+        - click >=7.0,<9
+        - cloudpickle <3
+        - databricks-cli >=0.8.7,<1
+        - docker-py >=4.0.0,<6  # [mlflow_variant != "skinny"]
+        - entrypoints <1
+        - flask <3  # [mlflow_variant != "skinny"]
+        - gitpython >=2.1.0,<4
         - gunicorn  # [not win and mlflow_variant != "skinny"]
         - numpy  # [mlflow_variant != "skinny"]
-        - pandas  # [mlflow_variant != "skinny"]
-        - prometheus_flask_exporter  # [mlflow_variant != "skinny"]
-        - packaging
-        - protobuf >=3.12.0
+        - pandas <2  # [mlflow_variant != "skinny"]
+        - prometheus_flask_exporter <1  # [mlflow_variant != "skinny"]
+        - packaging <22
+        - protobuf >=3.12.0,<5
         - python
-        - pytz
-        - pyyaml >=5.1
-        - querystring_parser  # [mlflow_variant != "skinny"]
-        - requests >=2.17.3
-        - scipy  # [mlflow_variant != "skinny"]
-        - sqlalchemy >=1.4.0  # [mlflow_variant != "skinny"]
-        - sqlparse >=0.3.1  # [mlflow_variant != "skinny"]
-        - waitress  # [win and mlflow_variant != "skinny"]
-        - importlib-metadata
+        - pytz <2023
+        - pyyaml >=5.1,<7
+        - querystring_parser <2  # [mlflow_variant != "skinny"]
+        - requests >=2.17.3,<3
+        - scipy <2  # [mlflow_variant != "skinny"]
+        - sqlalchemy >=1.4.0,<2  # [mlflow_variant != "skinny"]
+        - sqlparse >=0.4.0,<1
+        - waitress <3  # [win and mlflow_variant != "skinny"]
+        - importlib-metadata <5,>=3.7.0,!=4.7.0
         - openssl
-        # MLflow Pipelines dependencies
-        - scikit-learn >=1.0.*  # [mlflow_variant != "skinny"]
-        - pyarrow >=7.0.*  # [mlflow_variant != "skinny"]
-        - shap >=0.40.*  # [mlflow_variant != "skinny"]
-        # pandas-profiling version is not restricted to prevent the following jobs from failing:
-        # - linux linux_ppc64le_mlflow_variantdefaultpython3.10.____cpython
-        # - linux linux_ppc64le_mlflow_variantdefaultpython3.9.____cpython
-        - pandas-profiling  # [mlflow_variant != "skinny"]
-        - ipython >=7.0.*  # [mlflow_variant != "skinny"]
-        - markdown >=3.3.*  # [mlflow_variant != "skinny"]
-        - Jinja2 >=3.0.*  # [mlflow_variant != "skinny"]
```

Comment:
@xhochy Thanks for the update. Can we keep the dependencies for MLflow Pipelines?

Reply:
@harupy I would rather keep them as a separate package/outpt `mlflow-pipelines` (this is how we deal with extras in all other packages). I can do this as part of this PR.

Reply:
@xhochy To confirm, users who want to use MLflow Pipelines need to run the following command, right?

```bash
conda install mlflow-pipelines  # not `conda install mlflow`
```

Reply:
We internally discussed this. Let's create `mlflow-pipelines`.